server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeTests.java

```diff
@@ -189,39 +185,58 @@ public void testDiscoveryNodeToXContent() {
             "test-hostname",
             "test-hostaddr",
             transportAddress,
-            Collections.singletonMap("test-attr", "val"),
+            Map.of("test-attr", "val"),
             DiscoveryNodeRole.roles(),
             Version.CURRENT,
             withExternalId ? "test-external-id" : null
         );
 
-        final String jsonString = Strings.toString(node, randomBoolean(), randomBoolean());
-        final Map<String, Object> topLevelMap = XContentHelper.convertToMap(XContentType.JSON.xContent(), jsonString, randomBoolean());
-
-        assertThat(topLevelMap.toString(), topLevelMap.size(), equalTo(1));
-        assertTrue(topLevelMap.toString(), topLevelMap.containsKey("test-id"));
-
-        @SuppressWarnings("unchecked")
-        final Map<String, Object> detailsMap = (Map<String, Object>) topLevelMap.get("test-id");
+        assertThat(Strings.toString(node, true, false), equalTo(String.format(Locale.ROOT, """
```

Comment:
Nit: We can use `Strings.format` here without specifying the locale.

Reply:
I was assuming this it not allowed by the check-style.

https://github.com/elastic/elasticsearch/blob/a00a85cd8f5a5e98d89357a6700e7c3de78f690d/build-tools-internal/src/main/resources/forbidden/es-server-signatures.txt#L140-L142

Seems this is okay to do in the test code

Reply:
`String#format` is forbidden; `Strings#format` is good.

Reply:
Okay, I just confirmed this is enforced: https://gradle-enterprise.elastic.co/s/unbjcotx5tcs4

Reply:
Missed that, thanks all!

Reply:
I opened #93336